pkg/kpackcompat/clientset.go

```diff
@@ -0,0 +1,90 @@
+package kpackcompat
+
+import (
+	"fmt"
+
+	"github.com/pivotal/kpack/pkg/apis/build"
+	"github.com/pivotal/kpack/pkg/client/clientset/versioned/typed/build/v1alpha2"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/discovery"
+
+	// load credential helpers
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+const (
+	KpackGroupVersionV1alpha2 = build.GroupName + "/v1alpha2"
+)
+
+type ClientsetInterface interface {
+	Discovery() discovery.DiscoveryInterface
+	KpackV1alpha2() v1alpha2.KpackV1alpha2Interface
+}
+
+type ClientSet struct {
+	*discovery.DiscoveryClient
+	v1alpha2Client v1alpha2.KpackV1alpha2Interface
+}
+
+func (c *ClientSet) KpackV1alpha2() v1alpha2.KpackV1alpha2Interface {
+	return c.v1alpha2Client
+}
+
+func (c *ClientSet) Discovery() discovery.DiscoveryInterface {
+	if c == nil {
+		return nil
+	}
+	return c.DiscoveryClient
+}
+
+func NewForConfig(c *rest.Config) (ClientsetInterface, error) {
```

Comment:
instead of implementing our own NewClientset method that returns our own trimmed down ClientSet can we call the real `kpack.NewForConfig` and have a wrapper/adapter that upconverts where necessary? That avoids us copying all this RateLimiter/Burst logic here.